src/pages/iou/steps/IOUConfirmPage/IOUConfirmRequest.js

```diff
@@ -0,0 +1,151 @@
+import React, {Component} from 'react';
+import {View, TextInput} from 'react-native';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import ONYXKEYS from '../../../../ONYXKEYS';
+import styles from '../../../../styles/styles';
+import Text from '../../../../components/Text';
+import ButtonWithLoader from '../../../../components/ButtonWithLoader';
+import themeColors from '../../../../styles/themes/default';
+import {getDisplayOptionsFromParticipants} from '../../../../libs/OptionsListUtils';
+import OptionsList from '../../../../components/OptionsList';
+
+const propTypes = {
+    // Callback to inform parent modal of success
+    onConfirm: PropTypes.func.isRequired,
+
+    // IOU amount
+    iouAmount: PropTypes.string.isRequired,
+
+    // callback to update comment from IOUModal
+    onUpdateComment: PropTypes.func,
+
+    // Selected currency from the user
+    // remove eslint disable after currency symbol is available
+    // eslint-disable-next-line react/no-unused-prop-types
+    selectedCurrency: PropTypes.string.isRequired,
+
+    // comment value from IOUModal
+    comment: PropTypes.string,
+
+    // Selected participants from IOUMOdal with login
+    participants: PropTypes.arrayOf(PropTypes.shape({
+        login: PropTypes.string.isRequired,
+        alternateText: PropTypes.string,
+        hasDraftComment: PropTypes.bool,
+        icons: PropTypes.arrayOf(PropTypes.string),
+        searchText: PropTypes.string,
+        text: PropTypes.string,
+        keyForList: PropTypes.string,
+        isPinned: PropTypes.bool,
+        isUnread: PropTypes.bool,
+        reportID: PropTypes.number,
+        participantsList: PropTypes.arrayOf(PropTypes.object),
```

Comment:
I;m a bit confused as to why we need all this participant data? `hasDraftComment`,`searchText`, ect seems completely separate from the list of user emails that we need to pass to the API?

Reply:
Can't we make do with just the participant emails at this point?

Reply:
@Julesssss  Are you suggesting to just receive the participant logins in the `IOUConfirmRequest `page, and use the participants from Onyx and filter using those logins? 

Because just the logins wouldn't suffice and I'd need to access `icons`, `alternateText` and `text` for the participants as well. 

For this view: 

<img width="200" alt="Screenshot 2021-04-07 at 15 12 08" src="https://user-images.githubusercontent.com/19339044/113843802-a221db00-97b3-11eb-97c1-41519b2683f6.png">

Reply:
In that case, we could simplify passing just logins for the user as a prop. And then inside the IOUConfirmRequest, I could just filter out participants using the logins. Just an extra bit of logic inside IOUConfirmRequest but should simplify what we pass around as data. Please let me know if this solution sounds good to you! 

Also, should I make this PR Open, or just wait till another PR resolves (for the mergeCollection issue) so it could get a more finished review?

Reply:
Ah, so the participants object contains both the user data and the init state of the participants component? That makes more sense.

> Are you suggesting to just receive the participant logins in the IOUConfirmRequest page, and use the participants from Onyx and filter using those logins?

No, I was just a bit confused as to why we need all of those fields. Let's keep as is, but can we try to remove the unnecessary fields from the `participants` object prop? We only need the data that you mention here: `'I'd need to access icons, alternateText and text'`

Reply:
Also, should I make this PR Open, or just wait till another PR resolves (for the mergeCollection issue) so it could get a more finished review?

It's fine as WIP for now, I wasn't planning a full review just yet but might do later today if I have any remaining time.

Reply:
> Also, should I make this PR Open, or just wait till another PR resolves (for the mergeCollection issue) so it could get a more finished review?
> 
> It's fine as WIP for now, I wasn't planning a full review just yet but might do later today if I have any remaining time.

Got you! Will do that. You can take your time, I was wondering if it was a worry that this PR is on hold for so long. Just wanted a confirmation at this point. Haha

Reply:
Ah, don't worry about the '7 day' thing, this was a large and complex issue. This issue is a higher priority than the average issue, but it's the external blocker that we're awaiting.

Reply:
If all goes well though, that blocker will be resolved today 🤞

Reply:
Ahh all good!